fix: the polaris config relation non-daemon thread should stop when application fails to start.

shuiqingliu opened the pull request:
## PR Type
Bugfix
<!--
Bugfix.
Feature.
 -->

## Describe what this PR does for and how you did.
the polaris config relation non-daemon thread should stop when application fails to start.

## Adding the issue link (#xxx) if possible.
#1032
<!--
fixes #
 -->

## Note

## Checklist

- [x] Add information of this PR to CHANGELOG.md in root of project.
- [x] Add documentation in javadoc or comment below the PR if necessary.

## Checklist (Optional)

- [x] Will pull request to branch of 2020.0.
- [x] Will pull request to branch of 2022.0.

shuiqingliu committed:
fix: the polaris config relation non-daemon thread should stop when application fails to start.

SkyeBeFreeman committed:
begin 1.13.0.

shuiqingliu committed:
Merge remote-tracking branch 'upstream/2021.0' into feature-long-pulling-thread-stop

shuiqingliu committed:
feat: bump polaris version from 1.14.0 to 1.15.0-SNAPSHOT

SkyeBeFreeman committed:
Merge branch '2021.0' into feature-long-pulling-thread-stop

SkyeBeFreeman commented on spring-cloud-starter-tencent-polaris-config/src/main/java/com/tencent/cloud/polaris/config/listener/PolarisConfigNonDaemonThreadStopListener.java:
```diff
@@ -0,0 +1,50 @@
+ * @author shuiqingliu
+ * @since 2023/8/29
+ **/
+public class PolarisConfigNonDaemonThreadStopListener implements ApplicationListener<ApplicationEvent> {
```
这个类能不能放在context包下？这样所有失败的启动，都可以执行这一块。

shuiqingliu replied:
已处理

shuiqingliu committed:
fix: Register a destroy hook when the application event trigger

if we register a destroy hook for RemoteConfigFileRepo on the class instance , the static thread pool may be terminated by the bootstrap context closing ,Therefore,it's better to register the destroy hook when the  application is prepared, or when the application fails to start, in order to properly destroy resources

shuiqingliu committed:
Merge branch 'feature-long-pulling-thread-stop' of github.com:shuiqingliu/spring-cloud-tencent into feature-long-pulling-thread-stop

shuiqingliu committed:
Merge branch '2021.0' into feature-long-pulling-thread-stop

shuiqingliu committed:
fix: move the RemoteConfigFileRepo destroy hook to the Polaris config module.

shuiqingliu committed:
Merge branch 'feature-long-pulling-thread-stop' of github.com:shuiqingliu/spring-cloud-tencent into feature-long-pulling-thread-stop
